build-tools/packages/build-cli/src/handlers/askFunctions.ts

```diff
@@ -32,34 +29,8 @@ export const askForReleaseType: StateHandlerFunction = async (
 ): Promise<boolean> => {
 	if (testMode) return true;
 
-	const { bumpType: inputBumpType, context, releaseVersion } = data;
-
-	const currentBranch = await context.gitRepo.getCurrentBranchName();
-	const currentVersion = releaseVersion;
-	const bumpedMajor = bumpVersionScheme(currentVersion, "major");
-	const bumpedMinor = bumpVersionScheme(currentVersion, "minor");
-	const bumpedPatch = bumpVersionScheme(currentVersion, "patch");
-
-	// If an bumpType was set in the handler data, use it. Otherwise set it as the default for the branch. If there's
-	// no default for the branch, ask the user.
-	let bumpType = inputBumpType ?? getDefaultBumpTypeForBranch(currentBranch);
-	if (inputBumpType === undefined) {
-		const selectedBumpType: VersionBumpType = await rawlist({
-			choices: [
-				{ value: "major", name: `major (${currentVersion} => ${bumpedMajor.version})` },
-				{ value: "minor", name: `minor (${currentVersion} => ${bumpedMinor.version})` },
-				{ value: "patch", name: `patch  (${currentVersion} => ${bumpedPatch.version})` },
-			],
-			message: `The current branch is '${currentBranch}'. The default bump type for that branch is '${bumpType}', but you can change it now if needed.`,
-		});
-		bumpType = selectedBumpType;
-		// eslint-disable-next-line require-atomic-updates
-		data.bumpType = selectedBumpType;
-	}
-
-	if (bumpType === undefined) {
-		throw new Error(`bumpType is undefined.`);
-	}
```

Comment:
All this logical basically does nothing because inputBumpType can't be undefined based on its type. This also seems redundant with the logic which computes the inputBumpType. Logic has been simplified to use the already provided bump type (which will lazily prompt if needed)